clientlibs/js/src/functions/init.ts

```diff
@@ -0,0 +1,52 @@
+import { Interfaces, Types } from '../identifiers';
+import fetchDataService from '../common/fetchDataService';
+import convertMapToObj from '../common/convertMapToObj';
+
+export default async function init(
+  url: string,
+  userId: string,
+  token: string,
+  group?: Map<string, Array<string>>,
+  workingGroup?: Map<string, string>,
+): Promise<Interfaces.IUser> {
+  try {
+    let data: any = {
+      id: userId,
+    };
+
+    if (group) {
+      if (!(group instanceof Map)) {
+        throw new Error('Group type should be Map<string, Array<string>>');
+      }
+      const groupObj = convertMapToObj(group);
+      data = {
+        ...data,
+        groupObj
+      }
+    }
+
+    if (workingGroup) {
+      if (!(workingGroup instanceof Map)) {
+        throw new Error('Working group type should be Map<string, string>');
+      }
+      const workingGroupObj = convertMapToObj(workingGroup);
+      data = {
+        ...data,
+        workingGroupObj
+      }
+    }
+
+    const response = await fetchDataService(url, token, data, Types.REQUEST_TYPES.POST);
+    if (response.status) {
+      return {
+        id: userId,
+        group,
+        workingGroup
```

Comment:
@YashilDepani  Can we use the group and workingGroup which are getting in the response to forward it to caller of this function? I think that would be better instead of sending what we got in the argument.

Reply:
I am not able to use group and workingGroup of response as the function return expects IUSER interface and getting error as non assignable.

Reply:
I have updated the code.